aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -628,7 +628,16 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
     // Naive, composite implementation defined here.
 
     // Scale q, k before matmul for stability see https://tinyurl.com/sudb9s96 for math
-    const auto scaling_factor = sdp::calculate_scale(query_, scale).sqrt();
+    bool is_negative_scaling = false;
+    c10::SymFloat scaling_factor;
+    if (scale.has_value()) {
+      is_negative_scaling  = scale.value() < 0;
+      scaling_factor = c10::SymFloat(std::abs(scale.value()));
+    } else {
+      scaling_factor = c10::SymFloat(1.0) / (c10::SymFloat(query_.sym_size(-1)).sqrt());
+    }
+    scaling_factor = scaling_factor.sqrt();
```

Comment:
Are you still able to invoke `sdp::calculate_scale` without reimplementing its logic here?

Reply:
updated.